### PR TITLE
Fix scripts not running at Interface start after reloading all scripts

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Issue: https://github.com/kasenvr/project-athena/issues/122

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
#### Test Plan

1. Repro the problem.
- A previous version of Interface.
- Running Scripts dialog, press the "Reload All" button.
- Restart Interface.
There will be no scripts running.

2. Check that problem is fixed.
- Repeat the tests in 1 with the PR version of Interface. 
The scripts should stay running each Interface run.

**Comment by MarcusLlewellyn** [comment]
Unofficial PR build for windows here: [ProjectAthena-Alpha-PR162-5a91de0.exe](https://drive.google.com/file/d/1dWopYhZOaFjQtCb5kdMp2hPX3GCFdFKM/view?usp=sharing)

File: libraries/script-engine/src/ScriptEngines.cpp [hunk 1]
```diff
@@ -412,7 +412,7 @@ void ScriptEngines::stopAllScripts(bool restart) {
             continue;
         }
 
-        bool isOverrideScript = it.key().toString().compare(this->_defaultScriptsOverride.toString());
+        bool isOverrideScript = it.key().toString().compare(this->_defaultScriptsOverride.toString()) == 0;
```
Comment (HifiExperiments) [inline libraries/script-engine/src/ScriptEngines.cpp]: god this is confusing

**Comment by SilverfishVR** [comment]
Switching back an forth between K2 RC1 and the PR build over the last few days, It definitely seems fixed. Thank You, this was really starting to annoy me.

Q: Who closes the issue (#122) , the author or the fixer?

**Comment by kasenvr** [comment]
Either way is fine, I would wait to close an issue only once the fix is merged though.

**Pull request merged**